src/DynamicRupture/FrictionLaws/RateAndState.h

```diff
@@ -10,363 +10,443 @@ namespace seissol::dr::friction_law {
+    // TODO: Check if this is different for fast and slow velocity weakening
+    real tmp = 0.5 / this->drParameters.rs_sr0 * exp(localStateVariable[0] / a[ltsFace][0]) *
+               this->slipRateMagnitude[ltsFace][0];
+    assert(!std::isnan(tmp) && "nonConvergence RS Newton");
+  }
+
+  void calcSlipRateAndTraction(std::array<real, numPaddedPoints> const& stateVarReference,
+                               std::array<real, numPaddedPoints> const& localSlipRate,
                                std::array<real, numPaddedPoints>& localStateVariable,
-                               std::array<real, numPaddedPoints>& normalStress,
-                               std::array<real, numPaddedPoints>& TotalShearStressYZ,
-                               std::array<real, numPaddedPoints>& tmpSlip,
-                               real deltaStateVar[numPaddedPoints],
+                               std::array<real, numPaddedPoints> const& normalStress,
+                               std::array<real, numPaddedPoints> const& absoluteShearStress,
                                FaultStresses& faultStresses,
                                unsigned int timeIndex,
-                               unsigned int ltsFace);
-
-  /*
-   * resample output state variable
-   */
-  void resampleStateVar(real deltaStateVar[numPaddedPoints], unsigned int ltsFace);
-
-  // output time when shear stress is equal to the dynamic stress after rupture arrived
-  // currently only for linear slip weakening
-  void saveDynamicStressOutput(unsigned int face);
-
-  // set to zero since only required for Thermal pressure
-  virtual void hookSetInitialP_f(std::array<real, numPaddedPoints>& P_f, unsigned int ltsFace);
-
-  // empty since only required for Thermal pressure
-  virtual void hookCalcP_f(std::array<real, numPaddedPoints>& P_f,
-                           FaultStresses& faultStresses,
-                           bool saveTmpInTP,
-                           unsigned int timeIndex,
-                           unsigned int ltsFace);
+                               unsigned int ltsFace) {
+    for (int pointIndex = 0; pointIndex < numPaddedPoints; pointIndex++) {
+      // SV from mean slip rate in tmp
+      localStateVariable[pointIndex] =
+          static_cast<Derived*>(this)->updateStateVariable(pointIndex,
+                                                           ltsFace,
+                                                           stateVarReference[pointIndex],
+                                                           this->deltaT[timeIndex],
+                                                           localSlipRate[pointIndex]);
+
+      // update LocMu for next strength determination, only needed for last update
+      this->mu[ltsFace][pointIndex] = static_cast<Derived*>(this)->updateMu(
+          ltsFace, pointIndex, localStateVariable[pointIndex]);
+      real strength = this->mu[ltsFace][pointIndex] * normalStress[pointIndex];
```

Comment:
1) n_stress = min(ZERO, P - P_f) since #341
2) I would say strength should be positive so `strength = -this->mu[ltsFace][pointIndex] * normalStress[pointIndex];`
(and remove the minus signs later on)